api/proto/device/v1alpha1/gpu.proto

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package nvidia.nvsentinel.v1alpha1;
+
+option go_package = "github.com/nvidia/nvsentinel/api/gen/go/device/v1alpha1;v1alpha1";
+
+import "google/api/annotations.proto";
+import "google/api/http.proto";
+import "google/protobuf/timestamp.proto";
+
+// ==========================================
+// Resource Definitions
+// ==========================================
+
+// Gpu represents a single GPU resource in Kubernetes.
+message Gpu {
+  // name is the unique logical identifier of the GPU resource.
+  string name = 1;
+
+  // spec defines the identity and desired attributes of the GPU resource.
+  GpuSpec spec = 2;
+
+  // status contains the most recently observed state of the GPU resource.
+  // This data may lag slightly behind the actual on-device state.
+  GpuStatus status = 3;
+}
+
+// GpuList is a collection of GPU resources.
+message GpuList {
+  // items is the list of GPU resources.
+  repeated Gpu items = 1;
+}
+
+// GpuSpec describes the identity and desired attributes of the GPU.
+message GpuSpec {
+  // id is the physical hardware UUID of this GPU.
+  // Format: 'GPU-<hex-string>' (e.g., 'GPU-a1b2c3d4-e5f6-a7b8-c9d0-e1f2a3b4c5d6').
+  string id = 1;
+
+  // node_name is the name of the host node where this GPU is physically attached.
+  string node_name = 2;
```

Comment:
Since the API is going to be node local, I'm not sure we want to include the node name. Should we instead start with an empty spec and we can add properties to it as we make progress?

Reply:
where will you get the data for this? This variable is not relevant.